flow/project.py

```diff
@@ -4877,11 +4955,11 @@ def operation_function(job):
         except KeyError:
             raise KeyError(f"Unknown operation '{args.operation}'.")
 
-        group = self._groups[args.operation]
-        aggregate_store = self._group_to_aggregator[group]
-        for aggregate in aggregate_store.values():
-            if self._is_selected_aggregate(aggregate, aggregates):
-                operation_function(*aggregate)
+        for aggregate_id, aggregate, group in self._generate_selected_aggregate_groups(
+            selected_aggregates=aggregates,
+            selected_groups={self._groups[args.operation]},
```

Comment:
Selects just a single group -- this should be efficient for the iteration logic in `_generate_selected_aggregate_groups` because it will heavily filter its iterations. It should only iterate over the aggregate store for this group, and only handle the provided group.